src/lib/crypto/CMakeLists.txt

```diff
@@ -135,7 +135,9 @@ endif(WITH_BOTAN)
 include_directories(${INCLUDE_DIRS})
 
 add_library(${PROJECT_NAME} OBJECT ${SOURCES})
-target_link_libraries(${PROJECT_NAME} ${CRYPTO_LIBS})
+if(NOT CMAKE_VERSION VERSION_LESS "3.12")
```

Comment:
How about:
```
if(CMAKE_VERSION VERSION_GREATER "3.11")
```
The above is more confusing to look at. If accepted, please change all instances.

Reply:
I would use `VERSION_GREATER_EQUAL` if possible, but that unfortunately that requires at least https://cmake.org/cmake/help/v3.7/release/3.7.html, hence this (semantically equivalent) workaround.

`CMAKE_VERSION VERSION_GREATER "3.11"` would not be sufficient as it also matches 3.11.4. I could use `CMAKE_VERSION VERSION_GREATER "3.11.99"` if you think that is easier to understand?

Reply:
You're right, they have widely distributed point releases from time to time (i.e. CMake 3.0.2) ... 

I wish `VERSION_GREATER_EQUAL` existed much earlier.

Your original change is probably the least of all evils given my workaround had a flaw - leave it as is.

Maybe just add a reference to the comment in `src/lib/CMakeLists.txt` to explain the existence of the if-else block, because it looks very strange without the relevant context. (apply to all affected files)